rasa/shared/core/domain.py

```diff
@@ -1789,38 +1799,117 @@ def validate(mapping: Dict[Text, Any], form_name: Text, slot_name: Text) -> None
+                f"{DOCS_URL_SLOTS} for more information."
             )
 
         validations = {
             str(SlotMapping.FROM_ENTITY): ["entity"],
             str(SlotMapping.FROM_INTENT): ["value"],
             str(SlotMapping.FROM_TRIGGER_INTENT): ["value"],
             str(SlotMapping.FROM_TEXT): [],
+            str(SlotMapping.CUSTOM): [],
         }
 
         mapping_type = mapping.get("type")
         required_keys = validations.get(mapping_type)
 
         if required_keys is None:
             raise InvalidDomain(
-                f"Your form '{form_name}' uses an invalid slot mapping of type "
+                f"Your domain uses an invalid slot mapping of type "
                 f"'{mapping_type}' for slot '{slot_name}'. Please see "
-                f"{DOCS_URL_FORMS} for more information."
+                f"{DOCS_URL_SLOTS} for more information."
             )
 
         for required_key in required_keys:
             if mapping.get(required_key) is None:
                 raise InvalidDomain(
                     f"You need to specify a value for the key "
                     f"'{required_key}' in the slot mapping of type '{mapping_type}' "
-                    f"for slot '{slot_name}' in the form '{form_name}'. Please see "
-                    f"{DOCS_URL_FORMS} for more information."
+                    f"for slot '{slot_name}'. Please see "
+                    f"{DOCS_URL_SLOTS} for more information."
                 )
 
+    @staticmethod
+    def _get_ignored_intents(
+        mapping: Dict[Text, Any], domain: Domain, active_loop_name: Text,
+    ) -> List[Text]:
+        mapping_conditions = mapping.get("conditions")
+        active_loop_match = False
+        ignored_intents = []
+
+        if mapping_conditions:
+            for condition in mapping_conditions:
+                if condition.get("active_loop") == active_loop_name:
+                    active_loop_match = True
+                    break
+
+        if active_loop_match:
+            form_ignored_intents = domain.forms[active_loop_name].get(
+                IGNORED_INTENTS, []
+            )
+            if not isinstance(form_ignored_intents, list):
+                ignored_intents = [form_ignored_intents]
+            else:
+                ignored_intents = form_ignored_intents
+
+        return ignored_intents
+
+    @staticmethod
+    def intent_is_desired(
+        mapping: Dict[Text, Any], tracker: "DialogueStateTracker", domain: Domain,
+    ) -> bool:
+        """Checks whether user intent matches slot mapping intent specifications."""
+        mapping_intents = mapping.get("intent", [])
+        mapping_not_intents = mapping.get("not_intent", [])
+
+        active_loop_name = tracker.active_loop_name
+        if active_loop_name:
+            mapping_not_intents = set(
+                mapping_not_intents
+                + SlotMapping._get_ignored_intents(mapping, domain, active_loop_name)
```

Comment:
Doesn't this just match on `active_loop`? Do we know that the other conditions have matched?

Reply:
I don't think the other conditions are required for retrieving `ignored_intents` since this flag would apply regardless to all slots if the form is activated.